### PR TITLE
rofi-calc: init at 1.6

### PR DESCRIPTION
**Pull request by luc65r**
###### Motivation for this change

Add the rofi-calc plugin and add a way to add rofi plugins in a pure way as well.
Also adding @albakham as a maintainer since he's interested in helping to keep it up to date.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by luc65r** [comment]
related PR: https://github.com/rycee/home-manager/pull/1109

**Comment by wamserma** [comment]
@luc65r you might want to cherry-pick https://github.com/NixOS/nixpkgs/pull/79661/commits/e1efcdfd694afce1be996c6c4066d5a2564bdfbc here or move it to an individual PR

File: pkgs/applications/misc/rofi/wrapper.nix [hunk 1]
```diff
@@ -13,11 +13,20 @@ stdenv.mkDerivation {
     rm $out/bin
     mkdir $out/bin
     ln -s ${rofi-unwrapped}/bin/* $out/bin
+    rm $out/lib
```
Comment (teto) [inline pkgs/applications/misc/rofi/wrapper.nix]: this could be replaced with a symlinkJoin (lookup zathura's wrapper)

**Review by Ma27 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/rofi-calc/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  postPatch = ''
+    sed "s|qalc_binary = \"qalc\"|qalc_binary = \"${libqalculate}/bin/qalc\"|" -i src/calc.c
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Do live calculations in rofi!";
+    homepage = "https://github.com/svenstaro/rofi-calc";
+    license = licenses.mit;
+    mainteners = with maintainers; [ luc65r albakham ];
```
Comment (wamserma) [inline pkgs/applications/science/math/rofi-calc/default.nix]: and the property is also called `maintainers`

**Review by wamserma — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/rofi/wrapper.nix [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
+    rm -rf $out/bin
     mkdir $out/bin
     ln -s ${rofi-unwrapped}/bin/* $out/bin
 
     rm $out/bin/rofi
     makeWrapper ${rofi-unwrapped}/bin/rofi $out/bin/rofi \
       --prefix XDG_DATA_DIRS : ${hicolor-icon-theme}/share \
-      ${if theme != null then ''--add-flags "-theme ${theme}"'' else ""}
+      ${lib.optionalString (theme != null) ''--add-flags "-theme ${theme}"''} \
+      ${lib.optionalString (builtins.length plugins != 0) ''--add-flags "-plugin-path $out/lib/rofi"''}
```
Comment (rycee) [inline pkgs/applications/misc/rofi/wrapper.nix]:
```suggestion
      ${lib.optionalString (plugins != []) ''--add-flags "-plugin-path $out/lib/rofi"''}
```

**Review by garbas — APPROVED** [review]

**Pull request merged**